src/app/view/components/_common/dot-wizard/dot-wizard.component.ts

```diff
@@ -0,0 +1,212 @@
+                this.setValid(valid, index);
+            });
+        });
+    }
+
+    private consolidateValues(data: { [key: string]: string }, step: number): void {
+        if (this.stepsValidation[step] === true) {
+            this.wizardData = { ...this.wizardData, ...data };
+        }
+    }
+
+    private setDialogActions(): void {
+        this.dialogActions = {
+            accept: {
+                action: () => {
+                    this.getAcceptAction();
+                },
+                label: this.isLastStep()
+                    ? this.dotMessageService.get('send')
+                    : this.dotMessageService.get('next'),
+                disabled: true
+            },
+            cancel: {
+                action: () => {
+                    this.loadNextStep(-1);
+                },
+                label: this.dotMessageService.get('previous'),
+                disabled: true
+            }
+        };
+    }
+
+    private loadNextStep(next: number) {
+        this.currentStep += next;
+        this.focusFistFormElement();
+        this.updateTransform();
+        if (this.isLastStep()) {
+            this.dialogActions.accept.label = this.dotMessageService.get('send');
+            this.dialogActions.cancel.disabled = false;
+        } else if (this.isFirstStep()) {
+            this.dialogActions.cancel.disabled = true;
+            this.dialogActions.accept.label = this.dotMessageService.get('next');
+        } else {
+            this.dialogActions.cancel.disabled = false;
+            this.dialogActions.accept.label = this.dotMessageService.get('next');
+        }
+        this.dialogActions.accept.disabled = !this.stepsValidation[this.currentStep];
+    }
+
+    private getAcceptAction(): void {
+        this.isLastStep() ? this.sendValue() : this.loadNextStep(1);
+    }
+
+    private isLastStep(): boolean {
+        return this.currentStep === this.componentsHost.length - 1;
+    }
+
+    private isFirstStep(): boolean {
+        return this.currentStep === 0;
+    }
+
+    private setValid(valid: boolean, step: number): void {
+        this.stepsValidation[step] = valid;
+        if (this.currentStep === step) {
+            this.dialogActions.accept.disabled = !valid;
+        }
+    }
+
+    private sendValue(): void {
+        this.dotWizardService.output$(this.wizardData);
+        this.close();
+    }
+
+    private updateTransform(): void {
+        this.transform = `translateX(${this.currentStep * 400 * -1}px)`;
+    }
+
+    private focusFistFormElement(): void {
+        let count = 0;
+        // need to wait dynamic component to load the form.
+        const interval = setInterval(() => {
+            const form: HTMLFormElement = this.componentsHost[
+                this.currentStep
+            ].viewContainerRef.element.nativeElement.parentNode.children[0].getElementsByTagName(
+                'form'
+            )[0];
+            if (form || count === 10) {
```

Comment:
why 10?

Reply:
just set it randomly to give up to 2 seconds to the UI to load